### PR TITLE
Separate Node build/test into separate Travis Job

### PR DESCRIPTION
**Pull request by cdrini**
Refactor. This should make all the python jobs run faster (no more npm i for every job!), as well as make the logs easier to read.

### Technical
- Also cleaned up those silly curly quotes
- Organized the package.json scripts a bit more (linting != testing)
- Reduced `make test` commands

### Stakeholders
@cclauss

### CODE REVIEW

**Comment by cdrini** [comment]
Before: https://travis-ci.com/github/internetarchive/openlibrary/builds/187858221
After: https://travis-ci.com/github/internetarchive/openlibrary/builds/187910550

Total run time: 8min → 7.5min
Total time (including queuing): 25min → 12min

Note python 3.9 is the bottle neck for run time; not sure why, but it takes ~8min compared to the other pythons ~1.5min .

**Comment by cdrini** [comment]
@cclauss How strongly are attached to testing on 3.9? Removing it would drop our Travis times to <3min 😅

File: .travis.yml [hunk 1]
```diff
@@ -8,43 +8,52 @@ dist: focal
...
+        - make test-py
         - pip install safety
         - safety check || true
-    - name: “Python 3.8 on focal”
-      python: "3.8"
-    - name: “Python 3.9-dev on focal”
-      python: "3.9-dev"
+    - name: Python 3.8 on focal
+      python: 3.8
+    - name: Python 3.9-dev on focal
```
Comment (cclauss) [inline .travis.yml:25]: Please revert the removal of quotes on floating point values on yaml data.  Next week Python 3.9 ships.  The release next year at this time will be 3.10.  Without quotes on floating point numbers in yaml 3.10 is indispensable from 3.1 and Python 3.10 will be less than, not greater than 3.9.
```python
yaml_data = """
three_one: 3.1
three_eight: 3.8
three_nine: 3.9
three_ten: 3.10
"""
import yaml
print(yaml.load(yaml_data))
```

**Comment by cclauss** [comment]
Python 3.9 ships on Monday.  The wheels for dependencies will ship quickly thereafter at 3.9 will be as fast as 3.8.  Given how hard it is to upgrade things (like Python 2.7.6!j , my strong preference would be that we go into production on 3.9.

File: .travis.yml (hunk 1 above)
Comment (cdrini) [reply .travis.yml:25]: Ahh, good point! Fixing

Comment (cdrini) [reply .travis.yml:25]: Done.

**Comment by cdrini** [comment]
That seems reasonable to me; but if we do everything correctly, switching to 3.9 should be _significantly_ easier :) It'll just be updating the docker file, and then a docker-build + deploy \o/ Yeah, once 3.9 is officially released, it'll likely be much faster.

**Review by cclauss — APPROVED** [review]
Nice!

**Pull request merged**